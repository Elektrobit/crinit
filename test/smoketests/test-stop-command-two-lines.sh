# shellcheck shell=bash
# SPDX-License-Identifier: MIT
#
# smoketest for crinit stop command execution
#

setup() {
    crinit_config_setup
    if [ -e /tmp/test-stop_command_two_lines.out ]; then
        rm /tmp/test-stop_command_two_lines.out
    fi
}

run() {
    crinit_daemon_start "${SMOKETESTS_CONFDIR}"/demo.series
    sleep 3

    if ! "${BINDIR}"/crinit-ctl stop stop_command_two_lines; then
        echo "crinit-ctl stop failed"
        return 1
    fi

    sleep 1

    out="/tmp/test-stop_command_two_lines.out"
    sample="${CMDPATH}"/test-"${SMOKETESTS_NAME}".out
    compare_output "$sample" "$out"
}

teardown() {
    # Terminate crinit daemon
    crinit_daemon_stop
}
