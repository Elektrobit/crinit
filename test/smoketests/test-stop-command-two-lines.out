Hello, world from STOP_COMMAND.
This second line is echoed after the first.
